subiquity/server/controllers/filesystem.py

```diff
@@ -335,11 +386,62 @@ async def v2_reset_POST(self) -> StorageResponseV2:
         self.model.reset()
         return await self.v2_GET()
 
-    async def v2_guided_POST(self, data: GuidedChoice) -> StorageResponseV2:
+    async def v2_deprecated_guided_POST(self, data: GuidedChoice) \
+            -> StorageResponseV2:
         log.debug(data)
-        self.guided(data)
+        self.guided(GuidedChoiceV2.from_guided_choice(data))
         return await self.v2_GET()
 
+    async def v2_guided_GET(self) -> GuidedStorageResponseV2:
+        """Acquire a list of possible guided storage configuration scenarios.
+        Results are sorted by the size of the space potentially available to
+        the install."""
+
+        scenarios = []
+        install_min = self.calculate_suggested_install_min()
+
+        for disk in self.get_guided_disks(with_reformatting=True):
+            reformat = GuidedStorageTargetReformat(disk_id=disk.id)
+            scenarios.append((disk.size, reformat))
+
+        for disk in self.get_guided_disks(with_reformatting=False):
+            if len(disk.partitions()) < 1:
```

Comment:
Can you explain why you are doing this with a comment? I think I see but it would be nice to be clear.

Reply:
Done, see new commits